### PR TITLE
Adds tooltips for pneumatic cannon, changes powerfist's tooltip slightly and fixes some bugs about it.

### PR DESCRIPTION
**Pull request by Helg2**
## About The Pull Request
Adds tooltips for screwdriver and wrench with it's pressure level. It'll look like this:
image
And an image of how it look's at the moment just to compare:
image

Also changed powerfist's tooltip slightly:
image
And another image for comparison again:
image

You weren't able to put wrench in it, because it combined both LMB and RMB tool_acts, deleted the part with RMB as screwdriver uses LMB.

And for some reason you could put air tanks in pie guns and adjust its pressure level. Like is there even a purpose of it? It'll just hiss.
## Changelog
:cl:
qol: added tooltips for pneumatic cannon. Also changed powerfist's tooltips slightly.
fix: fixed that you can't put wrench in pneumatic cannon.
fix: fixed that you can manipulate with pie gun as if it's a normal pneumatic cannon (adjust air tank, wrench it's output level's and etc.)
/:cl:

### CODE REVIEW

File: code/game/objects/items/pneumaticCannon.dm [hunk 1]
```diff
@@ -82,21 +90,19 @@
...
-	switch(pressureSetting)
-		if(1)
-			pressureSetting = 2
-		if(2)
-			pressureSetting = 3
-		if(3)
-			pressureSetting = 1
-	to_chat(user, span_notice("You tweak \the [src]'s pressure output to [pressureSetting]."))
+	pressureSetting = pressureSetting >= HIGH_PRESSURE ? LOW_PRESSURE : pressureSetting + 1
+	balloon_alert(user, "output level set to [pressureSetting]")
```
Comment (Mothblocks) [inline code/game/objects/items/pneumaticCannon.dm:97]: Should be giving an actual word here, like "low/medium/high"

Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm:97]: Should i change it for powerfist too then?

Comment (Mothblocks) [reply code/game/objects/items/pneumaticCannon.dm:97]: Would be fine

File: code/game/objects/items/pneumaticCannon.dm [hunk 2]
```diff
@@ -93,7 +93,13 @@
 	if(needs_air == FALSE)
 		return
 	playsound(src, 'sound/items/ratchet.ogg', 50, TRUE)
-	pressureSetting = pressureSetting >= HIGH_PRESSURE ? LOW_PRESSURE : pressureSetting + 1
+	switch(pressureSetting)
+		if("low")
```
Comment (Mothblocks) [inline code/game/objects/items/pneumaticCannon.dm:97]: Use `LOW_PRESSURE`, `MID_PRESSURE`, `HIGH_PRESSURE`

File: code/game/objects/items/powerfist.dm [hunk 3]
```diff
@@ -42,12 +42,18 @@
...
-		. += span_notice("Can be removed with a screwdriver.")
+		. += span_notice("Can be removed with a <b>screwdriver</b>.")
 
-	. += span_notice("Use a wrench to change the valve strength. Current strength at [fist_pressure_setting].")
+	. += span_notice("Use a <b>wrench</b> to change the valve strength. Current strength is at <b>[fist_pressure_setting]</b> level.")
 
 /obj/item/melee/powerfist/wrench_act(mob/living/user, obj/item/tool)
-	fist_pressure_setting = fist_pressure_setting >= HIGH_PRESSURE ? LOW_PRESSURE : fist_pressure_setting + 1
+	switch(fist_pressure_setting)
+		if("low")
```
Comment (Mothblocks) [inline code/game/objects/items/powerfist.dm:51]: Same here

File: code/game/objects/items/pneumaticCannon.dm [hunk 4]
```diff
@@ -93,7 +93,13 @@
 	if(needs_air == FALSE)
 		return
 	playsound(src, 'sound/items/ratchet.ogg', 50, TRUE)
-	pressureSetting = pressureSetting >= HIGH_PRESSURE ? LOW_PRESSURE : pressureSetting + 1
+	switch(pressureSetting)
```
Comment (Mothblocks) [inline code/game/objects/items/pneumaticCannon.dm:96]: I'm skeptical you can just change these numbers and nothing else. Are you sure they are not treated as a number anywhere?

File: code/game/objects/items/pneumaticCannon.dm (hunk 2 above)
Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm:97]: Oh, right

File: code/game/objects/items/pneumaticCannon.dm (hunk 4 above)
Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm:96]: No idea how to do it then.

Comment (Mothblocks) [reply code/game/objects/items/pneumaticCannon.dm:96]: Make a proc that returns the current pressure setting as a string

**Review by Mothblocks — CHANGES_REQUESTED** [review]
The defines should stay as numbers, but you should be using a proc to get the pressure setting as a string

This code will break in your current setup:

image

File: code/game/objects/items/pneumaticCannon.dm (hunk 4 above)
Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm]: That'll do?

File: code/game/objects/items/powerfist.dm [hunk 5]
```diff
@@ -35,6 +35,15 @@
 	fire = 100
 	acid = 40
 
+/obj/item/melee/powerfist/proc/fist_pressure_setting_to_text(fist_pressure_setting)
```
Comment (Mothblocks) [inline code/game/objects/items/powerfist.dm:38]:
```suggestion
/obj/item/melee/powerfist/proc/pressure_setting_to_text(fist_pressure_setting)
```

No need to be redundant

File: code/game/objects/items/powerfist.dm [hunk 6]
```diff
@@ -35,6 +35,15 @@
...
+	switch(fist_pressure_setting)
+		if(LOW_PRESSURE)
+			. = "low"
```
Comment (Mothblocks) [inline code/game/objects/items/powerfist.dm:41]: `return`

File: code/game/objects/items/powerfist.dm [hunk 7]
```diff
@@ -35,6 +35,15 @@
 	fire = 100
 	acid = 40
 
+/obj/item/melee/powerfist/proc/fist_pressure_setting_to_text(fist_pressure_setting)
+	switch(fist_pressure_setting)
```
Comment (Mothblocks) [inline code/game/objects/items/powerfist.dm:39]: Add an `else` case here that calls `CRASH("Invalid pressure setting: [fist_pressure_setting]")

File: code/game/objects/items/pneumaticCannon.dm [hunk 8]
```diff
@@ -68,6 +68,15 @@
 /obj/item/pneumatic_cannon/CanItemAutoclick()
 	return automatic
 
+/obj/item/pneumatic_cannon/proc/pressureSetting_to_text(pressureSetting)
```
Comment (Mothblocks) [inline code/game/objects/items/pneumaticCannon.dm:71]: Same comments here, except make this `pressure_setting_to_text` as well, and the var `pressure_setting`. I know the var is pressureSetting but it should not be

File: code/game/objects/items/pneumaticCannon.dm [hunk 9]
```diff
@@ -93,13 +102,7 @@
...
 	playsound(src, 'sound/items/ratchet.ogg', 50, TRUE)
-	switch(pressureSetting)
-		if(LOW_PRESSURE)
-			pressureSetting = MID_PRESSURE
-		if(MID_PRESSURE)
-			pressureSetting = HIGH_PRESSURE
-		if(HIGH_PRESSURE)
-			pressureSetting = LOW_PRESSURE
+	pressureSetting = pressureSetting >= HIGH_PRESSURE ? LOW_PRESSURE : pressureSetting + 1
 	balloon_alert(user, "output level set to [pressureSetting]")
```
Comment (Mothblocks) [inline code/game/objects/items/pneumaticCannon.dm:106]: You didn't change this

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/objects/items/pneumaticCannon.dm (hunk 9 above)
Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm]: Can you explain

Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm]: You're not answering so i'll just set it on ready for review

Comment (Mothblocks) [reply code/game/objects/items/pneumaticCannon.dm]: This still gives the number and not the text

Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm]: Uhh, i don't quite understand why do you need the text outside of examine proc, and it changes it's regimes pretty fine:
image
I'm afraid i don't see an issue in point blank distance.
Also powerfist has same code for changing pressure settings so yeah.

Comment (Mothblocks) [reply code/game/objects/items/pneumaticCannon.dm]: This is the balloon alert. What does the balloon alert look like

File: code/game/objects/items/pneumaticCannon.dm [hunk 10]
```diff
@@ -68,6 +68,15 @@
 /obj/item/pneumatic_cannon/CanItemAutoclick()
 	return automatic
 
+/obj/item/pneumatic_cannon/proc/pressureSetting_to_text(pressureSetting)
+	switch(pressureSetting)
+		if(LOW_PRESSURE)
+			. = "low"
```
Comment (Mothblocks) [inline code/game/objects/items/pneumaticCannon.dm]: `return`

Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm]: Already done.

File: code/game/objects/items/pneumaticCannon.dm [hunk 11]
```diff
@@ -68,6 +68,15 @@
...
+		if(MID_PRESSURE)
+			. = "medium"
+		if(HIGH_PRESSURE)
```
[the remark this answers is not in the document]
Comment (Helg2) [reply code/game/objects/items/pneumaticCannon.dm:77]: Already done.

**Pull request merged**